common/src/main/java/com/viaversion/viaversion/protocols/base/BaseProtocol1_7.java

```diff
@@ -100,19 +102,19 @@ public void register() {
                         List<ProtocolPathEntry> protocols = null;
                         if (info.protocolVersion().higherThanOrEquals(closestServerProtocol) || Via.getPlatform().isOldClientsAllowed()) {
                             protocols = Via.getManager().getProtocolManager()
-                                .getProtocolPath(info.protocolVersion().getVersion(), closestServerProtocol.getVersion());
+                                .getProtocolPath(info.protocolVersion(), closestServerProtocol);
                         }
 
                         if (protocols != null) {
-                            if (protocolVersion == closestServerProtocol.getVersion() || protocolVersion == 0) { // Fix ServerListPlus
+                            if (protocolVersion.equals(closestServerProtocol) || !protocolVersion.isKnown()) { // Fix ServerListPlus
```

Comment:
This needs to check against 0, not all unknown I think as the SLP special case

Reply:
Sounds cursed, fixed